spec/Application/Processor/ProductLowestPriceBeforeDiscountProcessorSpec.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\PriceHistoryPlugin\Application\Processor;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Application\Processor\ProductLowestPriceBeforeDiscountProcessorInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntry;
+use Sylius\PriceHistoryPlugin\Infrastructure\Doctrine\ORM\ChannelPricingLogEntryRepositoryInterface;
+
+final class ProductLowestPriceBeforeDiscountProcessorSpec extends ObjectBehavior
+{
+    function let(ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository): void
+    {
+        $this->beConstructedWith($channelPricingLogEntryRepository);
+    }
+
+    function it_implements_product_lowest_price_processor_interface(): void
+    {
+        $this->shouldImplement(ProductLowestPriceBeforeDiscountProcessorInterface::class);
+    }
+
+    function it_sets_lowest_price_before_discount_to_null_if_currently_there_is_no_discount(
+        ChannelPricingInterface $channelPricing,
+    ): void {
+        $channelPricing->getOriginalPrice()->willReturn(null);
+        $channelPricing->getPrice()->willReturn(2100);
+        $channelPricing->setLowestPriceBeforeDiscount(null)->shouldBeCalled();
+
+        $this->process($channelPricing);
+    }
+
+    function it_sets_lowest_price_before_discount_to_lowest_price_found_in_the_given_period_if_currently_there_is_a_discount(
```

Comment:
We need to test a case when there are no log entries.

Reply:
Related to https://github.com/Sylius/PriceHistoryPlugin/pull/32#discussion_r1112890686